nixos/modules/services/networking/seafile.nix

```diff
@@ -293,6 +340,54 @@ in
             fi
           '';
         };
+
+        seaf-gc = {
+          description = "Seafile storage garbage collection";
+          conflicts = [ "seaf-server.service" "seahub.service" ];
+          after = [ "seaf-server.service" "seahub.service" ];
+          unitConfig.RequiresMountsFor = optionalDataDir;
+          onSuccess = [ "seaf-server.service" "seahub.service" ];
+          onFailure = [ "seaf-server.service" "seahub.service" ];
+          startAt = cfg.gcDates;
+          serviceConfig = securityOptions // {
+            User = cfg.user;
+            Group = cfg.group;
+            StateDirectory = "seafile";
+            RuntimeDirectory = "seafile";
+            LogsDirectory = "seafile";
+            ConfigurationDirectory = "seafile";
+            ReadWritePaths = optionalDataDir;
+            Type = "oneshot";
+          };
+          script = ''
+            if [ ! -f "${seafRoot}/server-setup" ]; then
+                echo "Server not setup yet, GC not needed" >&2
+                exit
+            fi
+
+            # checking for pending upgrades
+            installedMajor=$(cat "${seafRoot}/server-setup" | cut -d"-" -f1 | cut -d"." -f1)
+            installedMinor=$(cat "${seafRoot}/server-setup" | cut -d"-" -f1 | cut -d"." -f2)
+            pkgMajor=$(echo "${cfg.seafilePackage.version}" | cut -d"." -f1)
+            pkgMinor=$(echo "${cfg.seafilePackage.version}" | cut -d"." -f2)
+
+            if [[ $installedMajor == $pkgMajor && $installedMinor == $pkgMinor ]]; then
+               :
+            else
+                echo "Server not upgraded yet" >&2
```

Comment:
```suggestion
            if [[ $installedMajor != $pkgMajor && $installedMinor != $pkgMinor ]]; then
                echo "Server not upgraded yet" >&2
```

Reply:
Was kept to match the main script, but that isn't too useful I guess.